src/parser/Clauses.h

```diff
@@ -379,5 +379,47 @@ using InBoundClause = BoundClause;
 using OutBoundClause = BoundClause;
 using BothInOutClause = BoundClause;
 
+class NameLabelList {
+public:
+    NameLabelList() = default;
+
+    void add(std::string *label) {
+        labels_.emplace_back(label);
+    }
+
+    bool empty() const {
+        return labels_.empty();
+    }
+
+    std::size_t size() const {
+        return labels_.size();
+    }
+
+    const std::string* front() const {
+        return labels_.front().get();
+    }
+
+    std::vector<const std::string*> labels() const {
+        std::vector<const std::string*> labels;
+        labels.reserve(labels_.size());
+        for (const auto& it : labels_) {
+            labels.emplace_back(it.get());
+        }
+        return labels;
+    }
```

Comment:
Why not return labels_ directly? what's the purpose of this copying?

Reply:
Keep same with others like the YIeldClumns::columns.

Reply:
Not need it. We could fix that in the future.

Reply:
Ok, better to fix them together in another PR.